client/network/src/protocol/generic_proto/handler/notif_in.rs

```diff
@@ -156,16 +156,19 @@ impl ProtocolsHandler for NotifsInHandler {
 		&mut self,
 		(msg, proto): <Self::InboundProtocol as InboundUpgrade<NegotiatedSubstream>>::Output
 	) {
+		// If a substream already exists, we drop it and replace it with the new incoming one.
```

Comment:
What is the rationale for preferring the new substream over the old?

Reply:
It is forbidden to open a new substream if you already have one open. You're supposed to close the current one first.

I realized, however, that if the remote opens a substream, then closes it, then opens another one, then `inject_fully_negotiated_inbound` might be called a second time before we had the chance to realize that the first substream has been closed.

Consequently I went with simply always using with the most-recently opened substream.

Reply:
I'm not actually sure that this specific situation can happen with `yamux`, as it depends on how the `StreamMuxer` is implemented.

Reply:
I see, thanks.